### ISSUE TITLE
Zoom not working propertly when connectgaps= true

### ISSUE THREAD

**Issue by pawelciesielka** [plotly/plotly.js#6411]
I use [angular-plotly.js] version.
Graph with time series, draws line graph  with gaps not connected , with 100% zoom everything works fine we see line divided by gaps on 3 parts.
![obraz](https://user-images.githubusercontent.com/623162/208606647-93c410c4-6cb4-4f07-b5dd-a665412dbd9d.png)

but when you zoom, the first part (from begining to first gap)  is always visible on chart in original zoom 
![obraz](https://user-images.githubusercontent.com/623162/208607418-6aacffa3-a784-4dfe-9b0b-d1a397181bc4.png)

Config below:
```
layout= {title: 'Sg plot',
    shapes: [
      {
        type: 'line',
        xref: 'paper',
        x0: 0,
        y0: 70.0,
        x1: 1,
        y1: 70.0,
        line:{
          color: 'rgb(255, 0, 0)',
          width: 1,
          dash:'dot'
        }},
      {
        type: 'line',
        xref: 'paper',
        x0: 0,
        y0: 180.0,
        x1: 1,
        y1: 180.0,
        line:{
          color: 'rgb(255,174,0)',
          width: 1,
          dash:'dot'
        }
      },
      {
        type: 'line',
        xref: 'paper',
        x0: 0,
        y0: 90.0,
        x1: 1,
        y1: 90.0,
        line:{
          color: 'rgb(0,162,28)',
          width: 1,
          dash:'dot'
        }
      },
      {
        type: 'line',
        xref: 'paper',
        x0: 0,
        y0: 140.0,
        x1: 1,
        y1: 140.0,
        line:{
          color: 'rgb(0,162,28)',
          width: 1,
          dash:'dot'
        }

      }
    ]}

```
  ` data= [ { "x": [ "2022-12-13T12:01:00+01:00", "2022-12-13T12:01:00+01:00", "2022-12-13T12:06:00+01:00", "2022-12-13T12:11:00+01:00", "2022-12-13T12:16:00+01:00", "2022-12-13T12:21:00+01:00", "2022-12-13T12:26:00+01:00", "2022-12-13T12:31:00+01:00", "2022-12-13T12:36:00+01:00", "2022-12-13T12:41:00+01:00", "2022-12-13T12:46:00+01:00", "2022-12-13T12:51:00+01:00", "2022-12-13T12:56:00+01:00", "2022-12-13T13:01:00+01:00", "2022-12-13T13:06:00+01:00", "2022-12-13T13:11:00+01:00", "2022-12-13T13:16:00+01:00", "2022-12-13T13:21:00+01:00", "2022-12-13T13:26:00+01:00", "2022-12-13T13:31:00+01:00", "2022-12-13T13:36:00+01:00", "2022-12-13T13:41:00+01:00", "2022-12-13T13:46:00+01:00", "2022-12-13T13:51:00+01:00", "2022-12-13T13:56:00+01:00", "2022-12-13T14:01:00+01:00", "2022-12-13T14:06:00+01:00", "2022-12-13T14:11:00+01:00", "2022-12-13T14:16:00+01:00", "2022-12-13T14:21:00+01:00", "2022-12-13T14:26:00+01:00", "2022-12-13T14:31:00+01:00", "2022-12-13T14:36:00+01:00", "2022-12-13T14:41:00+01:00", "2022-12-13T14:46:00+01:00", "2022-12-13T14:51:00+01:00", "2022-12-13T14:56:00+01:00", "2022-12-13T15:01:00+01:00", "2022-12-13T15:06:00+01:00", "2022-12-13T15:11:00+01:00", "2022-12-13T15:16:00+01:00", "2022-12-13T15:21:00+01:00", "2022-12-13T15:26:00+01:00", "2022-12-13T15:31:00+01:00", "2022-12-13T15:36:00+01:00", "2022-12-13T15:41:00+01:00", "2022-12-13T15:46:00+01:00", "2022-12-13T15:51:00+01:00", "2022-12-13T15:56:00+01:00", "2022-12-13T16:01:00+01:00", "2022-12-13T16:06:00+01:00", "2022-12-13T16:11:00+01:00", "2022-12-13T16:16:00+01:00", "2022-12-13T16:21:00+01:00", "2022-12-13T16:26:00+01:00", "2022-12-13T16:31:00+01:00", "2022-12-13T16:36:00+01:00", "2022-12-13T16:41:00+01:00", "2022-12-13T16:46:00+01:00", "2022-12-13T16:51:00+01:00", "2022-12-13T16:56:00+01:00", "2022-12-13T17:01:00+01:00", "2022-12-13T17:06:00+01:00", "2022-12-13T17:11:00+01:00", "2022-12-13T17:16:00+01:00", "2022-12-13T17:21:00+01:00", "2022-12-13T17:26:00+01:00", "2022-12-13T17:31:00+01:00", "2022-12-13T17:36:00+01:00", "2022-12-13T17:41:00+01:00", "2022-12-13T17:46:00+01:00", "2022-12-13T17:51:00+01:00", "2022-12-13T17:56:00+01:00", "2022-12-13T18:01:00+01:00", "2022-12-13T18:06:00+01:00", "2022-12-13T18:11:00+01:00", "2022-12-13T18:16:00+01:00", "2022-12-13T18:21:00+01:00", "2022-12-13T18:26:00+01:00", "2022-12-13T18:31:00+01:00", "2022-12-13T18:36:00+01:00", "2022-12-13T18:41:00+01:00", "2022-12-13T18:46:00+01:00", "2022-12-13T18:51:00+01:00", "2022-12-13T18:56:00+01:00", "2022-12-13T19:01:00+01:00", "2022-12-13T19:06:00+01:00", "2022-12-13T19:11:00+01:00", "2022-12-13T19:16:00+01:00", "2022-12-13T19:21:00+01:00", "2022-12-13T19:26:00+01:00", "2022-12-13T19:31:00+01:00", "2022-12-13T19:36:00+01:00", "2022-12-13T19:41:00+01:00", "2022-12-13T19:46:00+01:00", "2022-12-13T19:51:00+01:00", "2022-12-13T19:56:00+01:00", "2022-12-13T20:01:00+01:00", "2022-12-13T20:06:00+01:00", "2022-12-13T20:11:00+01:00", "2022-12-13T20:16:00+01:00", "2022-12-13T20:21:00+01:00", "2022-12-13T20:26:00+01:00", "2022-12-13T20:31:00+01:00", "2022-12-13T20:36:00+01:00", "2022-12-13T20:41:00+01:00", "2022-12-13T20:46:00+01:00", "2022-12-13T20:51:00+01:00", "2022-12-13T20:56:00+01:00", "2022-12-13T21:01:00+01:00", "2022-12-13T21:06:00+01:00", "2022-12-13T21:11:00+01:00", "2022-12-13T21:16:00+01:00", "2022-12-13T21:21:00+01:00", "2022-12-13T21:26:00+01:00", "2022-12-13T21:31:00+01:00", "2022-12-13T21:36:00+01:00", "2022-12-13T21:41:00+01:00", "2022-12-13T21:46:00+01:00", "2022-12-13T21:51:00+01:00", "2022-12-13T21:56:00+01:00", "2022-12-13T22:01:00+01:00", "2022-12-13T22:06:00+01:00", "2022-12-13T22:11:00+01:00", "2022-12-13T22:16:00+01:00", "2022-12-13T22:21:00+01:00", "2022-12-13T22:26:00+01:00", "2022-12-13T22:31:00+01:00", "2022-12-13T22:36:00+01:00", "2022-12-13T22:41:00+01:00", "2022-12-13T22:46:00+01:00", "2022-12-13T22:51:00+01:00", "2022-12-13T22:56:00+01:00", "2022-12-13T23:01:00+01:00", "2022-12-13T23:06:00+01:00", "2022-12-13T23:11:00+01:00", "2022-12-13T23:16:00+01:00", "2022-12-13T23:21:00+01:00", "2022-12-13T23:26:00+01:00", "2022-12-13T23:31:00+01:00", "2022-12-13T23:36:00+01:00", "2022-12-13T23:41:00+01:00", "2022-12-13T23:46:00+01:00", "2022-12-13T23:51:00+01:00", "2022-12-13T23:56:00+01:00", "2022-12-14T00:01:00+01:00", "2022-12-14T00:06:00+01:00", "2022-12-14T00:11:00+01:00", "2022-12-14T00:16:00+01:00", "2022-12-14T00:21:00+01:00", "2022-12-14T00:26:00+01:00", "2022-12-14T00:31:00+01:00", "2022-12-14T00:36:00+01:00", "2022-12-14T00:41:00+01:00", "2022-12-14T00:46:00+01:00", "2022-12-14T00:51:00+01:00", "2022-12-14T00:56:00+01:00", "2022-12-14T01:01:00+01:00", "2022-12-14T01:06:00+01:00", "2022-12-14T01:11:00+01:00", "2022-12-14T01:16:00+01:00", "2022-12-14T01:21:00+01:00", "2022-12-14T01:26:00+01:00", "2022-12-14T01:31:00+01:00", "2022-12-14T01:36:00+01:00", "2022-12-14T01:41:00+01:00", "2022-12-14T01:46:00+01:00", "2022-12-14T01:51:00+01:00", "2022-12-14T01:56:00+01:00", "2022-12-14T02:01:00+01:00", "2022-12-14T02:06:00+01:00", "2022-12-14T02:11:00+01:00", "2022-12-14T02:16:00+01:00", "2022-12-14T02:21:00+01:00", "2022-12-14T02:26:00+01:00", "2022-12-14T02:31:00+01:00", "2022-12-14T02:36:00+01:00", "2022-12-14T02:41:00+01:00", "2022-12-14T02:46:00+01:00", "2022-12-14T02:51:00+01:00", "2022-12-14T02:56:00+01:00", "2022-12-14T03:01:00+01:00", "2022-12-14T03:06:00+01:00", "2022-12-14T03:11:00+01:00", "2022-12-14T03:16:00+01:00", "2022-12-14T03:21:00+01:00", "2022-12-14T03:26:00+01:00", "2022-12-14T03:31:00+01:00", "2022-12-14T03:36:00+01:00", "2022-12-14T03:41:00+01:00", "2022-12-14T03:46:00+01:00", "2022-12-14T03:51:00+01:00", "2022-12-14T03:56:00+01:00", "2022-12-14T04:01:00+01:00", "2022-12-14T04:06:00+01:00", "2022-12-14T04:11:00+01:00", "2022-12-14T04:16:00+01:00", "2022-12-14T04:21:00+01:00", "2022-12-14T04:26:00+01:00", "2022-12-14T04:31:00+01:00", "2022-12-14T04:36:00+01:00", "2022-12-14T04:41:00+01:00", "2022-12-14T04:46:00+01:00", "2022-12-14T04:51:00+01:00", "2022-12-14T04:56:00+01:00", "2022-12-14T05:01:00+01:00", "2022-12-14T05:06:00+01:00", "2022-12-14T05:11:00+01:00", "2022-12-14T05:16:00+01:00", "2022-12-14T05:21:00+01:00", "2022-12-14T05:26:00+01:00", "2022-12-14T05:31:00+01:00", "2022-12-14T05:36:00+01:00", "2022-12-14T05:41:00+01:00", "2022-12-14T05:46:00+01:00", "2022-12-14T05:51:00+01:00", "2022-12-14T05:56:00+01:00", "2022-12-14T06:01:00+01:00", "2022-12-14T06:06:00+01:00", "2022-12-14T06:11:00+01:00", "2022-12-14T06:16:00+01:00", "2022-12-14T06:21:00+01:00", "2022-12-14T06:26:00+01:00", "2022-12-14T06:31:00+01:00", "2022-12-14T06:36:00+01:00", "2022-12-14T06:41:00+01:00", "2022-12-14T06:46:00+01:00", "2022-12-14T06:51:00+01:00", "2022-12-14T06:56:00+01:00", "2022-12-14T07:01:00+01:00", "2022-12-14T07:06:00+01:00", "2022-12-14T07:11:00+01:00", "2022-12-14T07:16:00+01:00", "2022-12-14T07:21:00+01:00", "2022-12-14T07:26:00+01:00", "2022-12-14T07:31:00+01:00", "2022-12-14T07:36:00+01:00", "2022-12-14T07:41:00+01:00", "2022-12-14T07:46:00+01:00", "2022-12-14T07:51:00+01:00", "2022-12-14T07:56:00+01:00", "2022-12-14T08:01:00+01:00", "2022-12-14T08:06:00+01:00", "2022-12-14T08:11:00+01:00", "2022-12-14T08:16:00+01:00", "2022-12-14T08:21:00+01:00", "2022-12-14T08:26:00+01:00", "2022-12-14T08:31:00+01:00", "2022-12-14T08:36:00+01:00", "2022-12-14T08:41:00+01:00", "2022-12-14T08:46:00+01:00", "2022-12-14T08:51:00+01:00", "2022-12-14T08:56:00+01:00", "2022-12-14T09:01:00+01:00", "2022-12-14T09:06:00+01:00", "2022-12-14T09:11:00+01:00", "2022-12-14T09:16:00+01:00", "2022-12-14T09:21:00+01:00", "2022-12-14T09:26:00+01:00", "2022-12-14T09:31:00+01:00", "2022-12-14T09:36:00+01:00", "2022-12-14T09:41:00+01:00", "2022-12-14T09:46:00+01:00", "2022-12-14T09:51:00+01:00", "2022-12-14T09:56:00+01:00", "2022-12-14T10:01:00+01:00", "2022-12-14T10:06:00+01:00", "2022-12-14T10:11:00+01:00", "2022-12-14T10:16:00+01:00", "2022-12-14T10:21:00+01:00", "2022-12-14T10:26:00+01:00", "2022-12-14T10:31:00+01:00", "2022-12-14T10:36:00+01:00", "2022-12-14T10:41:00+01:00", "2022-12-14T10:46:00+01:00", "2022-12-14T10:51:00+01:00", "2022-12-14T10:56:00+01:00", "2022-12-14T11:01:00+01:00", "2022-12-14T11:06:00+01:00", "2022-12-14T11:11:00+01:00", "2022-12-14T11:16:00+01:00", "2022-12-14T11:21:00+01:00", "2022-12-14T11:26:00+01:00", "2022-12-14T11:31:00+01:00", "2022-12-14T11:36:00+01:00", "2022-12-14T11:41:00+01:00", "2022-12-14T11:46:00+01:00", "2022-12-14T11:51:00+01:00", "2022-12-14T11:56:00+01:00" ], "y": [ 0, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 111, 109, 108, 102, 101, 95, 78, 81, 85, 89, 100, 116, 124, 129, 123, 120, 116, 119, 140, 156, 157, 148, 132, 118, 104, 95, 102, 108, 115, 122, 132, 150, 161, 170, 174, 171, 157, 145, 137, 126, 133, 135, 127, 122, 121, 114, 100, 88, 85, 83, 84, 84, 81, 72, 59, 62, 67, 73, 76, 75, 77, 79, 79, 81, 84, 83, 82, 86, 87, 92, 105, 108, 112, 108, 100, 101, 116, 117, 121, 123, 118, 115, 113, 116, 118, 117, 110, 100, 101, 104, 110, 120, 122, 123, 119, 119, 119, 123, 126, null, null, null, null, null, null, null, null, 141, 147, 146, 143, 142, 139, 134, 131, 123, 104, 104, 97, 93, 97, 100, 104, 107, 108, 111, 120, 123, 121, 112, 105, 98, 95, 95, 98, 104, 105, 104, 102, 98, 96, 96, 96, 97, 94, 82, 84, 87, 90, 93, 91, 90, 88, 88, 90, 91, 92, 92, 90, 91, 94, 95, 95, 95, 96, 98, 98, 98, 97, 96, 97, 95, 93, 90, 88, 87, 89, 91, 97, 112, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 105, 110, 114, 116, 115, 112, 106, 99, 98, 100, 100, 102, 107, 110, 112, 109, 104, 104, 106, 108, 109, 111, 114, 114, 114, 111, 108, 108, 112, 114, 115, 120, 128, 139, 150, 161, 166, 171, 174, 171, 170, 167, 164, 164, 163, 159, 155, 150, 144, 138, 133, 128, 122, 117, 116, 117, 117, 115, 111, 112, 111, 110, 111, 109 ], "type": "scatter", "mode": "points", "line": { "shape": "points", "color": "rgb(152,152,152)", "width": 2 }, "connectgaps": false, "marker": { "symbol": "star-diamond", "color": "rgb(17, 157, 255)", "size": 8 } } ]`

**Comment by alexcjohnson** [comment]
Looks like a duplicate of #6404 and #6407

**Comment by alexcjohnson** [comment]
@pawelciesielka like #6407 this is fixed if you downgrade to v2.14.0 so it's almost certainly the same problem. I'll close this issue, we'll track progress in #6407, but before we close that we'll ensure the fix also works with your data.

**Comment by pawelciesielka** [comment]
@alexcjohnson it works as expected after downgrading to 2.14.0 , thanks for your help.
